scheduler/reconcile.go

```diff
@@ -527,12 +531,9 @@ func (a *allocReconciler) computeGroup(group string, all allocSet) bool {
 	// Final check to see if the deployment is complete is to ensure everything
 	// is healthy
 	if deploymentComplete && a.deployment != nil {
-		partOf, _ := untainted.filterByDeployment(a.deployment.ID)
-		for _, alloc := range partOf {
-			if !alloc.DeploymentStatus.IsHealthy() {
-				deploymentComplete = false
-				break
-			}
+		dstate := a.deployment.TaskGroups[group]
```

Comment:
Why did the logic for marking the deployment complete change? This means if the deployment status is healthy but desired total > Healthy allocs it won't mark the deployment as complete. 

Would be nice to make a reconciler test for this situation.

Reply:
Before there could be no failures so simply checking the scheduler didn't want to place anything new and all the existing placements where healthy was enough. Now there can be failures, so we have to check there aren't more desired placements and that we have reached our desired healthy count. I can add a test but will do to the follow up PR as this PR has some existing test failures fixed by the follow up.